proggy.cpp

```diff
@@ -92,25 +133,33 @@ std::string toLower(std::string const & data) {
     return ret;
 }
 
-Customer const applyDiscount(Customer const & cust, std::string comp) {
- 
-   auto ret = cust;
+//Customer const applyDiscount(Customer const & cust, std::string comp) {
+//** KOOS
+//** Pass in by ref, then sends a copy back??
```

Comment:
yeah; it's not the most optimal there. I tend to get the behavior in place for the interface and then consider how we want to handle whether we want to affect directly or pass copies around. Normally, I'd ask to clarify the preference and then roll with it. Passing copies can be easier for testing but changing in place makes it faster...